frb_codegen/src/generator/dart/ty.rs

```diff
@@ -16,6 +22,20 @@ pub trait TypeDartGeneratorTrait {
     fn structs(&self) -> String {
         "".to_string()
     }
+
+    fn get_context(&self) -> &TypeGeneratorContext;
+
+    fn is_type_shared(&self, ty: &IrType) -> bool {
+        self.get_context().ir_file.is_type_shared_by_safe_ident(ty)
+    }
+
+    fn get_private_prefix(&self) -> String {
```

Comment:
oh great here is a function for private, IIRC there are one or two places that can use this exact function

Reply:
I compacted all private methods in both dart and rust side.  I remembered  there are 2 conversations you mentioned using macro to compact the repeated `get_context()` code--- I didn't get it until 
f4c9c69.